packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -552,6 +555,31 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
         },
       );
 
+      registerServiceExtension(name: 'invokePreHotRestartCallbacks', callback: (Map<String, Object> params) async {
+        Future<void> invokeAndWait(DebugPreHotRestartCallback callback, String label) async {
+          developer.postEvent('preHotRestartCallback', <String, Object>{'label': label, 'finished': false});
+          try {
+            await Future<Object?>.value(callback());
+          } catch (error, stack) {
+            FlutterError.reportError(
+              FlutterErrorDetails(
+                exception: error,
+                stack: stack,
+                context: ErrorSummary('Failed to invoke preHotRestartCallback "$label"'),
```

Comment:
`context` needs to read correctly when following the word `thrown`, so something like `'while invoking the "$label" preHotRestartCallback'` or some such.

Reply:
(see https://master-api.flutter.dev/flutter/foundation/FlutterErrorDetails/context.html)

Reply:
Done 57ab854e84484a008307324f2dbf7ae991a2e40d